internal/core/src/segcore/ChunkedSegmentSealedImpl.cpp

```diff
@@ -178,8 +178,14 @@ ChunkedSegmentSealedImpl::LoadScalarIndex(const LoadIndexInfo& info) {
         return;
     }
 
-    scalar_indexings_[field_id] =
-        std::move(const_cast<LoadIndexInfo&>(info).cache_index);
+    if (info.index_params.find(index::INDEX_TYPE) != info.index_params.end() &&
```

Comment:
nit: avoid duplicate find:

if (auto it = info.index_params.find(index::INDEX_TYPE); it != info.index_params.end() && *it == ...)